### ISSUE TITLE
Shouldn't reach 100% when the tasks are not actually finished

### ISSUE THREAD

**Issue by HuStmpHrrr** [tqdm/tqdm#1640]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] visual output bug
- [ ] I have visited the [source website], and in particular
  read the [known issues]
- [ ] I have searched through the [issue tracker] for duplicates
- [ ] I have mentioned version numbers, operating system and
  environment, where applicable:
  ```python
  import tqdm, sys
  print(tqdm.__version__, sys.version, sys.platform)

4.67.1 3.13.0 (main, Oct 16 2024, 03:23:02) [Clang 18.1.8 ] linux
  ```

The title should be self-explanatory:

```
100%|███████████████████▍| 5068/5080 [07:28<00:10,  1.15it/s]
```

[source website]: https://github.com/tqdm/tqdm/
[known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
[issue tracker]: https://github.com/tqdm/tqdm/issues?q=

**Comment by Anas-Shakeel** [comment]
> The title should be self-explanatory:
> 
> ```
> 100%|███████████████████▍| 5068/5080 [07:28<00:10,  1.15it/s]
> ```

This happens because of the `f"{percentage:3.0f}%"` formatting, which rounds the percentage to the nearest whole number. So, when `frac * 100` results in something like `99.6%`, it gets rounded up to `100%` even though the iterations aren’t fully complete.

For example, (in your case): 
```py
total = 5080 # Total
n = 5068 # Current
```
So,
```py
frac = n / total   # 0.99763779527
percentage = frac * 100  # 99.7637795276
l_bar = f"{percentage:3.0f}%|" # 100%
```

### Possible Fix:
A possible fix would be to use `bar_format` parameter of `tqdm` to specify a custom bar string format with higher precision like `3.1f` or `3.2f`.
```py
for i in trange(0, 5080, bar_format='{desc}: {percentage:3.2f}%|{bar}{r_bar}'):
    # Your code...
    pass
```

### Output:
```sh
99.78%|█████████████████████████████████████████████████▉| 5069/5080 [00:00<00:00, 1897449.98it/s]
```

More about [tqdm parameters](https://github.com/tqdm/tqdm?tab=readme-ov-file#documentation)

**Comment by HuStmpHrrr** [comment]
rounding is just not correct. it should be flooring. if the formatter rounds automatically, then a floored number should be passed into manually,.

**Comment by Anas-Shakeel** [comment]
Yeah, that makes sense! Instead of letting the formatter round up, we should manually floor the percentage before formatting. Something like `math.floor(percentage)` would prevent it from hitting 100% too early.

Hopefully, the maintainers are open to fixing this!
